proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,532 @@
+                    "Add {obj_str} to the configure function that sets up the test harness.".
+                    format(obj_str=obj_str))
+            obj = getattr(self, obj_str)
+            setattr(m, obj_str, obj)
+
+    def configure(self):
+        """
+        Placeholder method to allow user to setup test harness.
+
+        The configure function must set the attributes:
+
+        prop_pack: property package parameter block
+
+        param_args: dictionary for property parameter arguments
+
+        set_default_scaling_dict: dictionary for default scaling
+            keys = (string name of variable, tuple index), values = scaling factor
+
+        solver: string name for solver, if None will use IDAES default
+
+        optarg: dictionary of solver options, if None will use IDAES default
+
+        default_state_values_dict: dictionary of state variables and their default values
+            keys = (string name of variable, tuple index), values = value of variable
+
+        default_initialize_results_dict: dictionary of property values for default state
+            keys = (string name of variable, tuple index), values = value of variable
+        """
+        pass
+
+    @pytest.fixture(scope='class')
+    def frame(self):
+        m = ConcreteModel()
+        self.configure_class(m)
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+        m.fs.properties = self.prop_pack()
+        m.fs.stream = m.fs.properties.build_state_block(
+            [0], default=self.param_args)
+
+        for (v_str, ind), sf in self.set_default_scaling_dict.items():
+            m.fs.properties.set_default_scaling(v_str, sf, index=ind)
+
+        return m
+
+    def test_default_value_initialize(self, frame):
+        m = frame
+
+        # # check state variables are at expected initial values
+        for (v_str, ind), val in m.default_state_values_dict.items():
+            var = getattr(m.fs.stream[0], v_str)
+            if not pytest.approx(val, rel=1e-3) == value(var[ind]):
+                raise Exception(
+                    "State variable {v_str} with index {ind} is expected to have a value of {val}, but it "
+                    "has a value of {val_t}. \nPlease update default_state_values_dict in the configure "
+                    "function that sets up the PropertyComponentTestHarness".format(
+                        v_str=v_str, ind=ind, val=val, val_t=value(var[ind])))
+
+        # touch properties
+        metadata = m.fs.properties.get_metadata().properties
+        for v_str in metadata.keys():
+            getattr(m.fs.stream[0], v_str)
+            try:
+                getattr(m.fs.stream[0], v_str)
+            except PropertyPackageError or PropertyNotSupportedError:
+                pass  # errors in metadata entries are tested in unit test harness
+
+        # scale model
+        calculate_scaling_factors(m)
+
+        # initialize
+        m.fs.stream.initialize(solver=m.solver, optarg=m.optarg)
+
```

Comment:
There are a lot of other things that should be checked here as well, most significantly that the model "state" has not changed (i.e. the set of fixed variables and activated constraints).